src/coreclr/src/tools/r2rdump/R2RDiff.cs

```diff
@@ -57,9 +57,54 @@ public void Run()
             DiffR2RSections();
             DiffR2RMethods();
 
-            Dictionary<string, ReadyToRunMethod> leftMethods = new Dictionary<string, ReadyToRunMethod>(_leftDumper.Reader.Methods
+            if (!_leftDumper.Reader.ReadyToRunHeader.Sections.TryGetValue(ReadyToRunSectionType.MethodDefEntryPoints, out ReadyToRunSection leftSection))
+            {
+                leftSection = new ReadyToRunSection();
+            }
+            if (!_rightDumper.Reader.ReadyToRunHeader.Sections.TryGetValue(ReadyToRunSectionType.MethodDefEntryPoints, out ReadyToRunSection rightSection))
+            {
+                rightSection = new ReadyToRunSection();
+            }
+
+            DiffMethodsForModule(leftSection, rightSection);
+
+            if (_leftDumper.Reader.Composite && _rightDumper.Reader.Composite)
+            {
+                HashSet<string> allComponentAssemblies = new HashSet<string>(_leftDumper.Reader.ManifestReferenceAssemblies);
+                allComponentAssemblies.UnionWith(_rightDumper.Reader.ManifestReferenceAssemblies);
+                foreach (string assemblyName in allComponentAssemblies.OrderBy(name => name))
+                {
+                    int leftIndex = _leftDumper.Reader.ManifestReferenceIndices[assemblyName];
+                    int rightIndex = _rightDumper.Reader.ManifestReferenceIndices[assemblyName];
+
+                    if (leftIndex < 0 ||  _leftDumper.Reader.ReadyToRunAssemblyHeaders[leftIndex].Sections.TryGetValue(ReadyToRunSectionType.MethodDefEntryPoints, out leftSection))
```

Comment:
It seems it should rather be
```suggestion
                    if (leftIndex < 0 ||  !_leftDumper.Reader.ReadyToRunAssemblyHeaders[leftIndex].Sections.TryGetValue(ReadyToRunSectionType.MethodDefEntryPoints, out leftSection))
```

Reply:
Ahh, good point, thanks for spotting!